proxy/proxy.go

```diff
@@ -924,10 +927,16 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 		p.tracing.setTag(ctx.proxySpan, ErrorTag, true)
 
 		// Check if the request has been cancelled or timed out
-		// The roundtrip error `err` may be different
+		// The roundtrip error `err` may be different:
+		// - for `Canceled` it could be either the same `context canceled` or `unexpected EOF` (net.OpError)
+		// - for `DeadlineExceeded` it is net.Error(timeout=true, temporary=true) wrapping this `context deadline exceeded`
 		if cerr := req.Context().Err(); cerr != nil {
 			ctx.proxySpan.LogKV("event", "error", "message", cerr.Error())
-			return nil, &proxyError{err: cerr, code: 499}
+			if cerr == stdlibcontext.Canceled {
```

Comment:
This now covers client cancellation and backend request timeout

Reply:
nice!